src/Components/Server/src/Builder/ComponentEndpointRouteBuilderExtensions.cs

```diff
@@ -29,16 +31,37 @@ public static ComponentEndpointConventionBuilder MapBlazorHub(this IEndpointRout
         }
 
         /// <summary>
-        /// Maps the SignalR <see cref="ComponentHub"/> to the path <see cref="ComponentHub.DefaultPath"/> and associates
+        /// Maps the Blazor <see cref="Hub" /> to the default path.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <param name="configureOptions">A callback to configure dispatcher options.</param>
+        /// <returns>The <see cref="ComponentEndpointConventionBuilder"/>.</returns>
+        public static ComponentEndpointConventionBuilder MapBlazorHub(this IEndpointRouteBuilder endpoints, Action<HttpConnectionDispatcherOptions> configureOptions)
+        {
+            if (endpoints == null)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            if (configureOptions == null)
+            {
+                throw new ArgumentNullException(nameof(configureOptions));
+            }
+
+            return new ComponentEndpointConventionBuilder(endpoints.MapHub<ComponentHub>(ComponentHub.DefaultPath, configureOptions));
```

Comment:
Is it possible to have this call through to one of the longer overloads? I always get nervous when there isn't a single implementation of these builder methods. We have a history of missing scenarios in tests.

Reply:
Similar comment about L286 - could you do a pass on these and try to reduce the duplication?

Reply:
Updated, there are only now two places where is terminal